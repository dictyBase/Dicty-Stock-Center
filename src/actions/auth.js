// @flow
import { dsctypes } from "constants/dsctypes"
import querystring from "querystring"
import oauthConfig from "utils/oauthConfig"
import { push } from "connected-react-router"

const {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT_SUCCESS,
  FETCH_USER_REQUEST,
  FETCH_USER_SUCCESS,
  FETCH_USER_FAILURE,
  FETCH_ROLE_REQUEST,
  FETCH_ROLE_SUCCESS,
  FETCH_ROLE_FAILURE,
  FETCH_NON_AUTH_ROLE_REQUEST,
  FETCH_NON_AUTH_ROLE_SUCCESS,
  FETCH_NON_AUTH_ROLE_FAILURE,
  FETCH_PERMISSION_REQUEST,
  FETCH_PERMISSION_SUCCESS,
  FETCH_PERMISSION_FAILURE,
} = dsctypes

type oauthArg = { query: string, provider: string, url: string }
type receiveLoginArg = { user: Object, token: string }

const authServer = process.env.REACT_APP_AUTH_SERVER
const apiServer = process.env.REACT_APP_API_SERVER
const fetchHeaderConfig = {
  headers: {
    "content-type": "application/vnd.api+json",
  },
}

const makeOauthConfig = ({ query, provider, url }: oauthArg) => {
  const parsed = querystring.parse(query.replace("?", ""))
  let body = `client_id=${oauthConfig[provider].clientId}&redirect_url=${url}`
  body += `&state=${parsed.state}&code=${parsed.code}`
  body += `&scopes=${oauthConfig[provider].scopes[0]}`
  let config = {
    method: "POST",
    headers: { "Content-Type": "application/x-www-form-urlencoded" },
    body: body,
  }
  const endpoint = `${authServer}/tokens/${provider}`
  return { config, endpoint }
}

const createErrorObj = (err: number, msg: string) => ({
  status: err,
  title: msg,
})

const requestLogin = (provider: string) => ({
  type: LOGIN_REQUEST,
  payload: {
    isFetching: true,
    provider: provider,
  },
})

const receiveLogin = ({ user, token }: receiveLoginArg) => ({
  type: LOGIN_SUCCESS,
  payload: {
    isFetching: false,
    token: token,
    user: user,
  },
})

const loginError = (error: Object) => ({
  type: LOGIN_FAILURE,
  payload: {
    isFetching: false,
    error,
  },
})

const receiveLogout = () => ({
  type: LOGOUT_SUCCESS,
  payload: {
    isFetching: false,
  },
})

const fetchUserRequest = () => ({
  type: FETCH_USER_REQUEST,
  payload: {
    isFetching: true,
  },
})

const fetchUserSuccess = (json: Object) => ({
  type: FETCH_USER_SUCCESS,
  payload: {
    isFetching: false,
    json,
  },
})

const fetchUserFailure = (error: Object) => ({
  type: FETCH_USER_FAILURE,
  payload: {
    error: error,
  },
})

const fetchRoleRequest = () => ({
  type: FETCH_ROLE_REQUEST,
  payload: {
    isFetching: true,
  },
})

const fetchRoleSuccess = (json: Object) => ({
  type: FETCH_ROLE_SUCCESS,
  payload: {
    isFetching: false,
    json,
  },
})

const fetchRoleFailure = (error: Object) => ({
  type: FETCH_ROLE_FAILURE,
  payload: {
    error: error,
  },
})

// Calls the API to get a token and
// dispatch actions along the way
const oAuthLogin = ({ query, provider, url }: oauthArg) => async (
  dispatch: Function,
) => {
  const { config, endpoint } = makeOauthConfig({ query, provider, url })
  try {
    dispatch(requestLogin(provider))
    dispatch(push("/load/auth"))
    const res = await fetch(endpoint, config)
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      if (res.ok) {
        const json = await res.json()
        dispatch(receiveLogin(json))
        await dispatch(fetchRoleInfo(json.user.data.id))
        dispatch(push("/mydsc"))
        // history.go(-3)
      } else if (res.status === 401) {
        // user has invalid credentials, redirect with notification
        dispatch(
          loginError(
            createErrorObj(
              res.status,
              `You are not an authorized user of dictyBase.
               Please sign in with proper credentials.`,
            ),
          ),
        )
        dispatch(push("/login"))
      } else {
        dispatch(loginError(createErrorObj(res.status, res.statusText)))
        dispatch(push("/error"))
      }
    } else {
      if (process.env.NODE_ENV !== "production") {
        console.error(res.statusText)
      }
      dispatch(loginError(createErrorObj(res.status, res.statusText)))
      dispatch(push("/error"))
    }
  } catch (error) {
    dispatch(loginError(createErrorObj(error.name, error.message)))
    dispatch(push("/error"))
  }
}

// Logs the user out
const logoutUser = () => (dispatch: Function) => {
  dispatch(receiveLogout())
}

// fetch user function that fetches data using async/await
// checks if header is correct, then either grabs data or displays error
// this is used to get a non-authenticated user's information
const fetchUserInfo = (userId: string) => async (dispatch: Function) => {
  try {
    dispatch(fetchUserRequest())
    const res = await fetch(`${apiServer}/users/${userId}`, fetchHeaderConfig)
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      const json = await res.json()
      if (res.ok) {
        dispatch(fetchUserSuccess(json))
        await dispatch(fetchNonAuthRoleInfo(json.data.id))
      } else {
        if (process.env.NODE_ENV !== "production") {
          printError(res, json)
        }
        dispatch(fetchUserFailure(json.errors[0]))
        dispatch(push("/error"))
      }
    } else {
      if (process.env.NODE_ENV !== "production") {
        console.error(res.statusText)
      }
      dispatch(fetchUserFailure(createErrorObj(res.status, res.statusText)))
      dispatch(push("/error"))
    }
  } catch (error) {
    dispatch(fetchUserFailure(error))
    dispatch(push("/error"))
    if (process.env.NODE_ENV !== "production") {
      console.error(`Network error: ${error.message}`)
    }
  }
}

// fetch roles function that fetches data using async/await
// checks if header is correct, then either grabs data or displays error
const fetchRoleInfo = (userId: string) => async (dispatch: Function) => {
  try {
    dispatch(fetchRoleRequest())
    const res = await fetch(
      `${apiServer}/users/${userId}/roles`,
      fetchHeaderConfig,
    )
    const contentType = res.headers.get("content-type")
    if (contentType && contentType.includes("application/vnd.api+json")) {
      const json = await res.json()
      if (res.ok) {
        if (json.data) {
          dispatch(fetchRoleSuccess(json))
          await dispatch(fetchPermissionInfo(json.data[0].id))
        }
      } else {
        if (process.env.NODE_ENV !== "production") {
          printError(res, json)
        }
        dispatch(fetchRoleFailure(json.errors[0].title))
      }
    } else {
      if (process.env.NODE_ENV !== "production") {
        console.error(res.statusText)
      }
      dispatch(fetchRoleFailure(createErrorObj(res.status, res.statusText)))
    }
  } catch (error) {
    dispatch(fetchUserFailure(error.toString()))
    if (process.env.NODE_ENV !== "production") {
      console.error(`Network error: ${error.message}`)
    }
  }
}

const fetchNonAuthRoleInfo = (userId: string) => ({
  types: [
    FETCH_NON_AUTH_ROLE_REQUEST,
    FETCH_NON_AUTH_ROLE_SUCCESS,
    FETCH_NON_AUTH_ROLE_FAILURE,
  ],
  url: `${apiServer}/users/${userId}/roles`,
  config: fetchHeaderConfig,
})

const fetchPermissionInfo = (roleId: string) => ({
  types: [
    FETCH_PERMISSION_REQUEST,
    FETCH_PERMISSION_SUCCESS,
    FETCH_PERMISSION_FAILURE,
  ],
  url: `${apiServer}/roles/${roleId}/permissions`,
  config: fetchHeaderConfig,
})

// helper function to print HTTP errors to console
// responses are structured in JSONAPI format
const printError = (res, json) => {
  console.error("HTTP Error")
  console.error(
    `HTTP Response: ${res.status}
    Title: ${json.errors[0].title}
    Detail: ${json.errors[0].detail}`,
  )
}

export {
  requestLogin,
  receiveLogin,
  loginError,
  fetchUserRequest,
  fetchUserSuccess,
  fetchUserFailure,
  fetchRoleRequest,
  fetchRoleSuccess,
  fetchRoleFailure,
  oAuthLogin,
  logoutUser,
  fetchUserInfo,
  fetchRoleInfo,
  fetchNonAuthRoleInfo,
  fetchPermissionInfo,
}
