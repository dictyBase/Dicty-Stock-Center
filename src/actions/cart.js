import { dsctypes } from "constants/dsctypes"
import { fees } from "constants/fees"

const { ADD_TO_CART, REMOVE_FROM_CART } = dsctypes
const { STRAIN_FEE, PLASMID_FEE, OTHER_FEE } = fees

const addItem = item => {
  let fee
  if (item.type === "strain") {
    fee = STRAIN_FEE
  } else if (item.type === "plasmid") {
    fee = PLASMID_FEE
  } else {
    fee = OTHER_FEE
  }
  return {
    type: ADD_TO_CART,
    fee: fee,
    item
  }
}

export const addToCart = item => {
  return dispatch => {
    dispatch(addItem(item))
  }
}

export const removeItem = id => {
  return (dispatch, getState) => {
    const { addedItems } = getState().cart
    const removeIndex = addedItems
      .map(itm => {
        return itm.id
      })
      .indexOf(id)
    dispatch({
      type: REMOVE_FROM_CART,
      removeIndex
    })
  }
}
