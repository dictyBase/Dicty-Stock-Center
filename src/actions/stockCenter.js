import React from 'react'
import types from 'constants'
import availability from 'fake-data/availability'

const { AVAILABILITY_FETCH_SUCCESS, STRAINS_FETCH_REQUEST, STRAINS_FETCH_SUCCESS } = types

// const requestAvailability = () => {
//     return {
//         type: AVAILABILITY_FETCH_REQUEST,
//         isFetching: true
//     }
// }
const strainList = {
    rows: [
        {
            availability: true,
            strain_descriptor: 'acaA-',
            strain_names: 'aca-, acaA-(r)',
            systematic_name: 'DG1108',
            id: 'DBS0252577'
        },
        {
            availability: true,
            strain_descriptor: 'abcB4-',
            strain_names: 'AK00010',
            systematic_name: 'DBS0350146',
            id: 'DBS0350146'
        }
    ],
    columns: [
        {
            property: 'availability',
            header: {
                label: 'Availability'
            }
        },
        {
            property: 'strain_descriptor',
            header: {
                label: 'Strain Descriptor'
            }
        },
        {
            property: 'strain_names',
            header: {
                label: 'Strain Names'
            }
        },
        {
            property: 'systematic_name',
            header: {
                label: 'Systematic Name'
            }
        },
        {
            property: 'id',
            header: {
                label: 'Strain ID'
            }
        },
        {
            cell: {
                format: (value, { rowData }) => (
                  <button
                    className="btn btn-primary"
                    onClick={ () => { console.log(`${JSON.stringify(rowData, null, 2)}`) } }>
                      Add to cart
                  </button>
                )
            },
            width: 200,
            visible: true
        }
    ]
}

const requestStrains = () => {
    return {
        type: STRAINS_FETCH_REQUEST
    }
}

const receiveStrains = (data) => {
    return {
        type: STRAINS_FETCH_SUCCESS,
        data
    }
}

const receiveAvailability = (data) => {
    return {
        type: AVAILABILITY_FETCH_SUCCESS,
        isFetching: false,
        availability: data.attributes.availability
    }
}

export const fetchStrainList = () => {
    return (dispatch) => {
        dispatch(requestStrains())
        setTimeout(() => {
            dispatch(receiveStrains(strainList))
        }, 1000)
    }
}

export const fetchAvailability = () => {
    return (dispatch) => {
        dispatch(receiveAvailability(availability.data))
    }
}
