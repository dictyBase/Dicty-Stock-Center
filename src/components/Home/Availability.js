// @flow
import React from "react"
import { useQuery } from "@apollo/react-hooks"
import Skeleton, { SkeletonTheme } from "react-loading-skeleton"
import { GET_STOCK_TOTALS } from "queries/queries"
import useStyles from "./homeStyles"

/**
 * Availability fetches and displays the current availability of strains and plasmids.
 */

const Availability = () => {
  const classes = useStyles()
  const { loading, error, data } = useQuery(GET_STOCK_TOTALS, {
    variables: {
      cursor: 0,
    },
  })

  if (loading)
    return (
      <SkeletonTheme color="#D3D3D3	" highlightColor="#DCDCDC">
        <Skeleton count={5} />
        &nbsp;
      </SkeletonTheme>
    )

  if (error)
    return (
      <div className={classes.panelGray}>
        Error fetching number of strains and plasmids
      </div>
    )

  return (
    <div className={classes.panelGray}>
      <h4>Strain & Plasmid Availability</h4>
      <h5>
        <strong>{data.listStrains.totalCount}</strong> Strains
      </h5>
      <h5>
        <strong>{data.listPlasmids.totalCount}</strong> Plasmids
      </h5>
    </div>
  )
}

export { Availability }
export default Availability
