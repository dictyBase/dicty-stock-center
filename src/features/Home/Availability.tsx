import React from "react"
import { useQuery } from "@apollo/client"
import PanelLoader from "./PanelLoader"
import { GET_STOCK_TOTALS } from "common/graphql/queries"
import useStyles from "./homeStyles"

/**
 * Availability fetches and displays the current availability of strains and plasmids.
 */

const Availability = () => {
  const classes = useStyles()
  const { loading, error, data } = useQuery(GET_STOCK_TOTALS, {
    variables: {
      cursor: 0,
    },
  })

  if (loading) return <PanelLoader skeletonCount={5} />

  if (error)
    return (
      <div className={classes.panelGray}>
        Error fetching number of strains and plasmids
      </div>
    )

  return (
    <div className={classes.panelGray}>
      <h4>Strain & Plasmid Availability</h4>
      <h5>
        <strong>{data.listStrains.totalCount}</strong> Strains
      </h5>
      <h5>
        <strong>{data.listPlasmids.totalCount}</strong> Plasmids
      </h5>
    </div>
  )
}

export default Availability
