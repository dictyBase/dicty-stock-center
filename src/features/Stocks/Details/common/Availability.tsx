import React from "react"
import { makeStyles } from "@material-ui/core/styles"
import Alert from "@material-ui/lab/Alert"
import AvailableDisplay from "./AvailableDisplay"
import { CartItem } from "common/types"

const useStyles = makeStyles((theme) => ({
  message: {
    padding: "0px",
  },
}))

type Props = {
  cartData: CartItem
  inStock: boolean
}

/**
 * Availability handles the display to indicate a stock's current
 * inventory status.
 */

const Availability = ({ cartData, inStock }: Props) => {
  const classes = useStyles()

  return (
    <React.Fragment>
      {inStock ? (
        <AvailableDisplay cartData={cartData} />
      ) : (
        <Alert
          classes={{ message: classes.message }}
          icon={false}
          severity="error">
          Currently unavailable at the DSC
        </Alert>
      )}
    </React.Fragment>
  )
}
export default Availability
