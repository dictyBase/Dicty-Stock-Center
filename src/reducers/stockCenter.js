import types from 'constants'

const {
    AVAILABILITY_FETCH_SUCCESS,
    AVAILABILITY_FETCH_REQUEST,
    AVAILABILITY_FETCH_FAILURE,
    STRAINS_FETCH_REQUEST,
    STRAINS_FETCH_SUCCESS,
    STRAINS_FETCH_FAILURE,
    STRAIN_FETCH_REQUEST,
    STRAIN_FETCH_SUCCESS,
    STRAIN_FETCH_FAILURE,
    STRAINS_SEARCH_REQUEST,
    STRAINS_SEARCH_SUCCESS,
    STRAINS_SEARCH_FAILURE,
    CLEAR_STRAINS,
    PLASMIDS_FETCH_REQUEST,
    PLASMIDS_FETCH_SUCCESS,
    PLASMIDS_FETCH_FAILURE,
    PLASMIDS_SEARCH_REQUEST,
    PLASMIDS_SEARCH_SUCCESS,
    PLASMIDS_SEARCH_FAILURE,
    CLEAR_PLASMIDS
} = types

const initialState = {
    availability: {
        isFetching: false
    },
    strainCatalog: {
        isFetching: false,
        data: [],
        links: {},
        meta: {
            pagination: {
                number: 1
            }
        }
    },
    strain: {
        isFetching: false
    },
    plasmidCatalog: {
        isFetching: false,
        data: [],
        links: {},
        meta: {
            pagination: {
                number: 1
            }
        }
    },
    plasmid: {
        isFetching: false
    }
}

const stockCenterReducer = (state = initialState, action) => {
    switch (action.type) {
    case AVAILABILITY_FETCH_REQUEST:
        return {
            ...state,
            availability: {
                isFetching: true
            }
        }
    case AVAILABILITY_FETCH_SUCCESS:
        return {
            ...state,
            availability: {
                isFetching: false,
                data: action.availability
            }
        }
    case AVAILABILITY_FETCH_FAILURE:
        return {
            ...state,
            availability: {
                isFetching: false,
                error: action.error
            }
        }
    case STRAINS_FETCH_REQUEST:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                isFetching: true
            }
        }
    case STRAINS_FETCH_SUCCESS:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                isFetching: false,
                links: action.links,
                meta: action.meta,
                data: state.strainCatalog.data.concat(action.data)
            }
        }
    case STRAINS_FETCH_FAILURE:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                isFetching: false,
                error: action.error
            }
        }
    case STRAINS_SEARCH_REQUEST:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                isFetching: true
            }
        }
    case STRAINS_SEARCH_SUCCESS:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                isFetching: false,
                data: action.data,
                links: action.links,
                meta: action.meta
            }
        }
    case STRAINS_SEARCH_FAILURE:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                isFetching: false,
                error: action.error
            }
        }
    case CLEAR_STRAINS:
        return {
            ...state,
            strainCatalog: {
                ...state.strainCatalog,
                data: [],
                meta: {
                    pagination: {
                        number: 1
                    }
                },
                links: {}
            }
        }
    case STRAIN_FETCH_REQUEST:
        return {
            ...state,
            strain: {
                ...state.strain,
                isFetching: true
            }
        }
    case STRAIN_FETCH_SUCCESS:
        return {
            ...state,
            strain: {
                isFetching: false,
                ...action.data
            }
        }
    case STRAIN_FETCH_FAILURE:
        return {
            ...state,
            strain: {
                ...state.strain,
                isFetching: false,
                error: action.error
            }
        }
    case PLASMIDS_FETCH_REQUEST:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                isFetching: true
            }
        }
    case PLASMIDS_FETCH_SUCCESS:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                isFetching: false,
                links: action.links,
                meta: action.meta,
                data: state.plasmidCatalog.data.concat(action.data)
            }
        }
    case PLASMIDS_FETCH_FAILURE:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                isFetching: false,
                error: action.error
            }
        }
    case PLASMIDS_SEARCH_REQUEST:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                isFetching: true
            }
        }
    case PLASMIDS_SEARCH_SUCCESS:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                isFetching: false,
                data: action.data,
                links: action.links,
                meta: action.meta
            }
        }
    case PLASMIDS_SEARCH_FAILURE:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                isFetching: false,
                error: action.error
            }
        }
    case CLEAR_PLASMIDS:
        return {
            ...state,
            plasmidCatalog: {
                ...state.plasmidCatalog,
                data: [],
                links: {},
                meta: {
                    pagination: {
                        number: 1
                    }
                }
            }
        }
    default:
        return state
    }
}

export default stockCenterReducer
