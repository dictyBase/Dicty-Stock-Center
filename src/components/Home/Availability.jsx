import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchAvailability } from 'actions/stockCenter'
import { PanelGray } from 'styles'

export class Availability extends Component {
    displayName = 'stock center availability'

    componentDidMount() {
        this.props.fetchAvailability()
    }
    render() {
        const { data } = this.props.availability
        return (
            <PanelGray>
                <h4>Strain & Plasmid Availability</h4>
                { data && data.map((item, index) => {
                    return (
                        <h5 key={ index }>
                            <strong>{ item.amount }</strong> { item.name }
                        </h5>
                    )
                }) }
            </PanelGray>
        )
    }
}

const mapStateToProps = state => {
    return {
        availability: state.stockCenter.availability
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchAvailability: () => {
            dispatch(fetchAvailability())
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Availability)
