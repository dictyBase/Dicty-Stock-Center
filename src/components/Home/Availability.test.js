import React from 'react'
import renderer from 'react-test-renderer'
import Availability from './Availability'
import * as stockCenterActions from '../../actions/stockCenter'

test('matching a snapshot of Availability', () => {
    const stockCenter = {
        availability: {
            type: 'data',
            id: '1',
            attributes: {
                availability: [
          { name: 'Strains', amount: 1927 },
          { name: 'Plasmids', amount: 882 },
          { name: 'Antibodies', amount: 12 },
          { name: 'cDNA library', amount: 1 },
          { name: 'Genomic library', amount: 1 }
                ]
            }
        }
    }
    const component = renderer.create(
    <Availability
      stockCenter={ stockCenter }
      stockCenterActions={ stockCenterActions }
    />
  )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})
