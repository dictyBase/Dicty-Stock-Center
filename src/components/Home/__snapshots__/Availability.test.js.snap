// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Availability 1`] = `
<div
  className="sc-itybZL dYOfed"
>
  <h4>
    Strain & Plasmid Availability
  </h4>
</div>
`;
